### ISSUE TITLE
Macro incorrectly assumed to be imported via `self` path.

### ISSUE THREAD

**Issue by Nashenas88** [rust-analyzer/rust-analyzer#9466]
```rust
pub mod trace {
    #[macro_export]
    macro_rules! trace {
        () => {}
    }

    pub struct A;
}

pub mod inner {
    use crate::trace::{self, A};

    pub fn abc() {
        let _a = A;
        trace!();
    }
}

fn main() {
    inner::abc();
}
```

If rust-analyzer's `checkOnSave` is disabled, then no error is produced for the usage of the `trace!` macro call in the `fn abc` above. Rust-analyzer sees the trace macro in scope, but this fails to compile because when using `self` in a mod path, it should only refer to the module and not the macro. In rust, the macro is only recognized if imported as `use crate::trace`, which is not the behavior seen in rust-analyzer.

**Comment by Veykril** [comment]
This was fixed in https://github.com/rust-analyzer/rust-analyzer/pull/9422 which will land next week.
